eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -33,36 +33,38 @@ jobs:
           build.args: ' -DTRANSPORT_PAHO=ON'
           test_type: 'iot'
           os: 'macos'
-        # Disable live testing for these matrix entries until we have something to
-        # test here.
-        # Linux_x64_with_samples:
-        #   Pool: azsdk-pool-mms-ubuntu-1804-general
-        #   OSVmImage: MMSUbuntu18.04
-        #   vcpkg.deps: 'curl[ssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        #   build.args: ' -DTRANSPO.RT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
-        # Win_x86_with_sampldes:
-        #   Pool: azsdk-pool-mms-win-2019-general
-        #   OSVmImage: MMS2019
-        #   vcpkg.deps: 'curl[winssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
-        #   CMAKE_GENERATOR: 'Visual Studio 16 2019'
-        #   CMAKE_GENERATOR_PLATFORM: Win32
-        #   build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=WIN32'
-        # Win_x64_with_samples:
-        #   Pool: azsdk-pool-mms-win-2019-general
-        #   OSVmImage: MMS2019
-        #   vcpkg.deps: 'curl[winssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
-        #   CMAKE_GENERATOR: 'Visual Studio 16 2019'
-        #   CMAKE_GENERATOR_PLATFORM: x64
-        #   build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=WIN32'
-        # MacOS_x64_with_samples:
-        #   Pool: Azure Pipelines
-        #   OSVmImage: 'macOS-10.15'
-        #   vcpkg.deps: 'curl[ssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-        #   build.args: ' -DTRANSPORT_CURL=ON  -DAZ_PLATFORM_IMPL=POSIX'
+        Linux_x64_with_samples:
+          Pool: azsdk-pool-mms-ubuntu-1804-general
+          OSVmImage: MMSUbuntu18.04
+          vcpkg.deps: 'curl[ssl]'
+          VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
+          os: 'linux'
```

Comment:
Do we want a os name specifically or could the image name work?

Reply:
I think it is the `os` name: `[14:53] Ahson Khan
Expanded: and(contains(' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX', '-DTRANSPORT_CURL=ON'), or(eq(Null, 'linux'), eq(Null, 'macos'))) Result: False`

Reply:
Also, this mirrors how the existing legs are currently configured, so I think it should work.

Reply:
We have `os name` on the other live test legs already.

And it is being used in this condition:
https://github.com/Azure/azure-sdk-for-c/blob/cf8f2b1fd768aa942836e98709d34ba0ced60d5e/eng/pipelines/templates/jobs/archetype-sdk-tests.yml#L94

If it makes sense to you @weshaggard  to improve that and use an existing field, that seems reasonable, and we could do that in a separate PR. I would leave it up to engsys folks like Daniel to drive that :)

Reply:
I generally like to avoid duplicating information so if OSVMImage would work here then that is more common and would be preferred.